### ISSUE TITLE
[Specs] Use MySQL to evaluate db migrations

### ISSUE THREAD

**Issue by dazza-codes** [dpn-admin/dpn-server#52]
The `api-v2-mysql-specs` branch can be used to run db migrations on MySQL.  The script in that branch at `db/mysql_migration_check.sh` can be used to check the migrations from `api-v1` to `api-v2`.

See https://github.com/dpn-admin/dpn-server/pull/53

The work-flow goes something like this:

``` sh
git checkout api-v2-mysql-specs
git rebase develop # any branch you like really
bundle install
bundle exec rake config
mysql -u root -p < db/mysql_init.sql
RAILS_ENV=test bundle exec rake db:drop
RAILS_ENV=test bundle exec rake db:create
RAILS_ENV=test bundle exec rake db:migrate
bundle exec rake # which runs specs
# if it all looks good, then have travis confirm it using
git push -f
```

The main-line of development is using SQLite; sometimes ActiveRecord migrations are not directly translated from SQLite to MySQL.  There are various pro/con points for developing with SQLite or MySQL.  These are some points in favor of developing with MySQL:
- ActiveRecord has some inconsistent behavior on SQLite vs. MySQL
- Approx. 4/5 DPN nodes use MySQL in production
- MySQL setup is easy given a linux-distro installation and running `db/mysql_init.sql` in the `api-v2-mysql-specs` branch
- Developing with MySQL should avoid last-minute problems with deployment
- When using MySQL, specs can be run without `--tag ~broken_in_ci` on travis.ci

**Comment by malakai97** [comment]
In a sense, by way of the PRs for this, these tests are already being performed by travis ci.

**Comment by dazza-codes** [comment]
Given the work from https://github.com/dpn-admin/dpn-server/pull/118, travis.ci can now run specs against 3 db platforms (sqlite, mysql, & postgres).  This should help to identify code bugs specific to each db platform.  It does not resolve specific migration problems when migrating production data from api_v1 to api_v2, which depend on live data migrations.  The travis.ci builds begin with a clean slate every time.

**Comment by dazza-codes** [comment]
Once the PR in https://github.com/dpn-admin/dpn-server/pull/121 (or something like it) is merged, this issue could be closed and some branches can be deleted.

**Comment by malakai97** [comment]
merged #121
